### PR TITLE
refactor: enum and comments

### PR DESCRIPTION
**Pull request by PaulFarault**
#### Which issue(s) this PR fixes

Fixes None

#### Additional comments

- refactor models
- propagate tests to cover all models
- improve db test to reset the db at each test
- add comments

#### Agreements

- [X] I hereby declare this contribution to be licensed under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0).
- [X] I hereby agree to grant [TOSIT](https://www.tosit.io/) a copyright license to use my contributions.

### CODE REVIEW

File: tdp/core/models/base.py [hunk 1]
```diff
@@ -7,14 +7,32 @@
...
     for k, v in vars(obj).items():
         if not k.startswith("_") and not any(hasattr(v, a) for a in excl):
             yield k, v
 
 
 class CustomBase:
+    """Custom base class for SQLAlchemy models.
+
+    Attributes:
+        __tablename__ (str): Table name.
```
Comment (rpignolet) [inline tdp/core/models/base.py:30]: I do not see this attribute in `CustomBase`.

File: tdp/core/models/test_db.py [hunk 2]
```diff
@@ -2,47 +2,98 @@
...
 
-@pytest.fixture(scope="session")
-def session_class():
+def empty_tables(session: Session):
+    table: Table
+    for table in reversed(Base.metadata.sorted_tables):
+        session.execute(table.delete())
+
+
+@pytest.fixture(scope="function")
```
Comment (rpignolet) [inline tdp/core/models/test_db.py:26]: The default scope is `function`, you can remove the scope.

File: tdp/core/models/test_db.py [hunk 3]
```diff
@@ -2,47 +2,98 @@
...
+from .deployment_log import DeploymentLog
+from .operation_log import OperationLog
+from .service_component_log import ServiceComponentLog
 
 logger = logging.getLogger("tdp").getChild("test_db")
 
 
-@pytest.fixture(scope="session")
-def session_class():
+def empty_tables(session: Session):
```
Comment (rpignolet) [inline tdp/core/models/test_db.py:20]: `clear_tables` is better. `empty` is generaly used to check if something is empty but not for removing items.

File: tdp/core/models/test_db.py [hunk 4]
```diff
@@ -2,47 +2,98 @@
...
+
+@pytest.fixture(scope="function")
+def session():
     engine = create_engine("sqlite+pysqlite:///:memory:", echo=True, future=True)
     Base.metadata.create_all(engine)
-    session_class = sessionmaker(bind=engine)
-    return session_class
+    session = sessionmaker(bind=engine)()
+
+    empty_tables(session)
```
Comment (rpignolet) [inline tdp/core/models/test_db.py:32]: Why do this before `yield` ? This is used to clear the database no ?

**Comment by PaulFarault** [comment]
All done @rpignolet. Thanks for your review!

**Review by rpignolet — APPROVED** [review]

**Pull request merged**